include/swift/AST/Module.h

```diff
@@ -951,7 +951,8 @@ inline SourceLoc extractNearestSourceLoc(const ModuleDecl *mod) {
 /// Collects modules that this module imports via `@_exported import`.
 void collectParsedExportedImports(const ModuleDecl *M,
                                   SmallPtrSetImpl<ModuleDecl *> &Imports,
-                                  llvm::SmallDenseMap<ModuleDecl *, SmallPtrSet<Decl *, 4>, 4> &QualifiedImports);
+                                  llvm::SmallDenseMap<ModuleDecl *, SmallPtrSet<Decl *, 4>, 4> &QualifiedImports,
+                                  bool DropNoDoc = false);
```

Comment:
I'm fine with this, but for extra flexibility have you also considered making this function take a predicate closure that takes an import and returns whether it should be dropped or not? That way we don't need to hardcode behavior for `@_nodoc` in the implementation.

Reply:
This function was effectively written for SymbolGraphGen in the first place, and only used transitively by anything that calls `getDisplayDecls` or `getTopLevelDeclsForDisplay` (which itself calls `getDisplayDecls`). We could change how this is structured, but SymbolGraphGen is still going to be the only one affected. 😅